### PR TITLE
Ja volunteer calendar fix

### PR DESCRIPTION
**Pull request by josearmandohf**

### CODE REVIEW

File: UnitTests/CMSWebTests/Areas/Manage/Models/VolunteerCommitmentsModelTest.cs [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+{
+    public class VolunteerCommitmentsModelTest
+    {
+        [Fact]
+        public void ShouldCalculateFirstSundayOfTheMonth()
+        {            
+            var FakeOrg = FakeOrganizationUtils.MakeFakeOrganization(RegistrationTypeCode.ChooseVolunteerTimes);            
+            var m = new VolunteerCommitmentsModel(FakeOrg.org.OrganizationId);            
+            var calculatedSunday = m.GetFirstSundayOfTheMonth(DateTime.Now.Year, DateTime.Now.Month);
+            Assert.Equal(DayOfWeek.Sunday, calculatedSunday.DayOfWeek);
```
Comment (brondavies) [inline UnitTests/CMSWebTests/Areas/Manage/Models/VolunteerCommitmentsModelTest.cs]: Use the Shouldly package for asserting valid data.  Examples exist in the other unit tests

File: UnitTests/CMSWebTests/Areas/Manage/Models/VolunteerCommitmentsModelTest.cs [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+using System.Text;
+using System.Threading.Tasks;
+using CmsData.Codes;
+using CmsWeb.Areas.Manage.Models;
+
+namespace CMSWebTests.Areas.Manage.Models
+{
+    public class VolunteerCommitmentsModelTest
+    {
+        [Fact]
```
Comment (brondavies) [inline UnitTests/CMSWebTests/Areas/Manage/Models/VolunteerCommitmentsModelTest.cs]: Change this to a `[Theory]` with inline data to define several specific test cases for various dates that may be calculated.

File: UnitTests/CMSWebTests/TestUtils/FakeOrganizationUtils.cs [hunk 3]
```diff
@@ -15,7 +17,7 @@ public class FakeOrganizationUtils
         public static Organization FakeOrganization { get; set; }
         public static int? OrgId { get; set; }
 
-        public static NewOrganizationModel MakeFakeOrganization()
+        public static NewOrganizationModel MakeFakeOrganization(int registrationType)
         {
```
Comment (jfescobar18) [inline UnitTests/CMSWebTests/TestUtils/FakeOrganizationUtils.cs]: Use the last version of MakeFakeOrganization here is an example
https://github.com/bvcms/bvcms/blob/develop/UnitTests/CMSWebTests/Areas/OnlineReg/Models/OnlineRegPerson/HelperTests.cs

File: UnitTests/CMSWebTests/Areas/Manage/Models/VolunteerCommitmentsModelTest.cs (hunk 1 above)
Comment (josearmandohf) [reply UnitTests/CMSWebTests/Areas/Manage/Models/VolunteerCommitmentsModelTest.cs]: ok

File: UnitTests/CMSWebTests/TestUtils/FakeOrganizationUtils.cs (hunk 3 above)
Comment (josearmandohf) [reply UnitTests/CMSWebTests/TestUtils/FakeOrganizationUtils.cs]: thanks!

File: UnitTests/CMSWebTests/app.config [hunk 4]
```diff
@@ -1,133 +1,146 @@
...
+  <connectionStrings>
+    <add name="CMS" connectionString="Data Source=(local);Integrated Security=True" />
+    <add name="Elmah.Sql" connectionString="Data Source=(local);Initial Catalog=ELMAH;Integrated Security=True" />
+  </connectionStrings>
+  <appSettings file="secrets.config">
+    <add key="PythonLibPath" value="c:\dev\bvcms\cmsweb\Lib" />
+    <add key="ClientSettingsProvider.ServiceUri" value="" />
+  </appSettings>
+  <system.web>
+    <membership defaultProvider="ClientAuthenticationMembershipProvider">
```
Comment (brondavies) [inline UnitTests/CMSWebTests/app.config]: Do we need this?

Comment (josearmandohf) [reply UnitTests/CMSWebTests/app.config]: removed

**Comment by brondavies** [comment]
> TeamCity Bvcms / Build [Build 1755](https://builds.tpsdb.com/viewLog.html?buildId=7807&buildTypeId=Bvcms_Build) outcome was **FAILURE**
> Summary: Tests failed: 13 (13 new), passed: 66 Build time: 00:19:13
It looks like the tests need to be fixed.  Let me know if you want help troubleshooting these

**Pull request merged**